### ISSUE TITLE
 OTP is correct and the time is not expired. It still shows the code has expired

### ISSUE THREAD

**Issue by hacker-0123** [opentrace-community/opentrace-android#25]
1. Before entering the OPT the message shows "Verification failed. Please try again"

2. Then after entering the OTP, message shows  "Wrong OTP entered"

Execption I get at console- **com.google.firebase.auth.FirebaseAuthInvalidCredentialsException: The sms code has expired. Please re-send the verification code to try again.**  -ERROR_SESSION_EXPIRED

**Comment by brianssendagire** [comment]
I'm having a similar problem. Did you find a solution for this?

**Comment by Hoangtho11188** [comment]
As my knowledge, you should enable phone authentication on Firebase first.

**Comment by Biplovkumar** [comment]
Also plz Setup of cloud function then set 
declare type SUPPORTED_REGIONS = "us-central1" | "us-east1" | "us-east4" | "europe-west1" | "europe-west2" | "asia-east2" | "asia-northeast1"; 
in gradle.properties.

**Comment by hacker-0123** [comment]
@Biplovkumar yeah given in the build cofig but still same. I would be grateful if you can send me your backend firebase config. Details in my mail address.

**Comment by vishaljoseph** [comment]
Hi, I'm having a similar issue. Were you guys able to solve it?

**Comment by hacker-0123** [comment]
Just need to fix firebase function properly.

**Comment by vishaljoseph** [comment]
I have set up cloud functions as elaborated in the README and they have even appeared on my firebase dashboard. It still doesn't seem to be working. Do I have to include any particular regions in my config.ts file? Any idea what I could have possibly missed?

**Comment by hacker-0123** [comment]
Send me  how it appears in your dashboard. Are u getting handshake key and
temporary id

**Comment by vishaljoseph** [comment]
**The logs:**

D/OnboardingActivity:  NOT-IDLE signInWithCredential:success
W/e.opentrace.st: Accessing hidden method Lcom/android/org/conscrypt/OpenSSLSocketImpl;->setAlpnProtocols([B)V (light greylist, reflection)
W/e.opentrace.st: Accessing hidden method Ljava/security/spec/ECParameterSpec;->setCurveName(Ljava/lang/String;)V (light greylist, reflection)
W/e.opentrace.st: Accessing hidden method Lcom/android/org/conscrypt/OpenSSLSocketImpl;->getAlpnSelectedProtocol()[B (light greylist, reflection)
D/TempIDManager:  NOT-IDLE [TempID] Error getting Temporary IDs
D/OnboardingActivity:  NOT-IDLE Retrieved Temporary ID successfully
W/Utils:  NOT-IDLE get handshake pin (failure): NOT_FOUND
E/OnboardingActivity:  NOT-IDLE Failed to retrieve HandShakePin NOT_FOUND
I/PhoneAuthProvider: Sms auto retrieval timed-out.

**Dashboard:**

![cloudfunc](https://user-images.githubusercontent.com/34835725/81001915-0df21a00-8e66-11ea-82d3-94b671c26eae.PNG)

**Comment by hacker-0123** [comment]
Seems fine to me, hope u have given same region in both client and server
side.

**Comment by vishaljoseph** [comment]
I had mistakenly put in a different region in the code and changing it seems to have solved the problem. Thanks a lot!

**Comment by vishaljoseph** [comment]
If I'm not wrong, the database location of my GCP Datastore should be in the same region as the above right?

**Comment by alexissinglaire** [comment]
> If I'm not wrong, the database location of my GCP Datastore should be in the same region as the above right?

@vishaljoseph : hi, do you have any issue in the app operation? if yes, maybe you can elaborate the detail here so we can help you to find the solution.

**Comment by Norbert-Mwebaze** [comment]
I keep on running into this issue is there a fix to it...... Thanks

**Comment by Norbert-Mwebaze** [comment]
OTP comes through but i get the message "Verification failed. Please try it again"

**Comment by capnmctavish** [comment]
Same issue, correct otp is entered still verification fails, timeout has also not occured.
But it succeeds when using the hardcoded numbers with preset OTP seamlessly

**Comment by purvajg** [comment]
did anyone find a fix for this?

**Comment by ghost** [comment]
I am having the same issue. When user request for Otp then OTP will automatically read and trying to verifying with the firebase then it shows **The sms verification code is expired** and when user enter manually then it worked and user can logged in.

**Comment by ahmad-alk** [comment]
Same issue here ☹️☹️

**Comment by ijunaidkhan** [comment]
same issue with my angular app, any solution?

https://stackoverflow.com/questions/72347766/angular-firebase-phone-authentication-gives-an-error-of-wrong-otp-even-it-is-co

**Comment by 696558586537** [comment]
Lebanon

**Comment by 696558586537** [comment]
pquxew@gmail.com
